### PR TITLE
Python builder API; clean up and additional operation details

### PR DESCRIPTION
**Pull request by abramvandergeest**
- adding detail/capabilities to input
- including a cml builder API in python (under tools)
- including a current version file for the python API to know which is the current version to use
- remove custom operation
- including subCML directory for examples of sub CMLs
- resolving conflict in sort operation with Steven's PR (accepting his changes)

### CODE REVIEW

**Comment by mellistibco** [comment]
let me know when you're done adding to the pr and i'll review.

**Comment by abramvandergeest** [comment]
Okay I am done pushing.  You can review.

File: subCMLs/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# SubCMLs
+
+Sub CMLs are CatalystML jsons that are intended to be run within another CML json with the runCML operation.  The usefully ness of this is that it allows you to recycle commonly reused combinations of operations.  Effectively making your own transferable operations.
```
Comment (mellistibco) [inline subCMLs/README.md]:
```suggestion
Sub CMLs are CatalystML jsons that are intended to be run within another CML json with the runCML operation.  The usefulness of this is that it allows you to recycle commonly used transformations.
```

Comment (mellistibco) [reply subCMLs/README.md]: Not really sure about the name `subCMLs`...

File: tools/python_builder/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Python Tool for building CatalystML
```
Comment (mellistibco) [inline tools/python_builder/README.md]:
```suggestion
# Python API for building CatalystML
```

File: versions/v0.1.5/Specification.MD [hunk 3]
```diff
@@ -66,9 +66,10 @@ General information describes several key:value pairs within the first level of
 
 The root level key value “input” is used to define the data that is passed into the specification.  Since multiple data objects can be passed to the specification “input” is a JSON list of objects containing the following key words:
 * type - [string]
-* dimensions - array]- if relevant
+* dimensions - [array]- if relevant
 * shape - [array]-  if relevant
-* label - [string] if labeling full input, [array] if providing labels for each component, [map] is proviging labels for compenents of a map
+* label - [string] if labeling full input, [array] if providing labels for each component using "_" to skip assigning labels if desired, [map] is proviging labels for compenents of a map.  Errors are thrown if labels are given for non-existant data references (i.e. arr[9] if the array is only lenth 5, or if key1 in {"key1":"label3"} doesn't exist).
```
Comment (mellistibco) [inline versions/v0.1.5/Specification.MD]:
```suggestion
* label - [string] if labeling full input, [array] if providing labels for each component using "_" to skip assigning labels if desired, [map] is providing labels for components of a map.  Errors are thrown if labels are given for non-existing data references (i.e. arr[9] if the array is only length 5, or if key1 in {"key1":"label3"} doesn't exist).
```

File: versions/v0.1.5/Specification.MD [hunk 4]
```diff
@@ -601,8 +603,8 @@ _Utils_
     * OutputType - [any]
     * Output - single string adds output to that variable, list of strings applies output of each input object to the corresponding outputlist item
 
-_Custom Operations_:
-* Language specific method for defining a custom operation (i.e. in go using go get)
+_Additional Operations_:
+* If further operations are needed.  Please suggest them through the Github Issue or PR  system.
```
Comment (mellistibco) [inline versions/v0.1.5/Specification.MD]: Not sure this statement should be part of the specification... Perhaps a better place would be a CONTRIBUTING.md in the root of the repo?

File: subCMLs/README.md (hunk 1 above)
Comment (abramvandergeest) [reply subCMLs/README.md]: I am open to other suggestions.

Comment (mellistibco) [reply subCMLs/README.md]: I'm also not sure we need a top level dir named `subCMLs` what else will be in this dir, besides this readme?

File: versions/v0.1.5/Specification.MD (hunk 4 above)
Comment (abramvandergeest) [reply versions/v0.1.5/Specification.MD]: I have made a CONTRIBUTING.md with a quick blurb we can improve in the future.

**Pull request merged**